tests/test-team/template-mgmt-component-tests/template-mgmt-manage-templates-page.component.ts

```diff
@@ -0,0 +1,298 @@
+    await manageTemplatesPage.clickCreateTemplateButton();
+    const chooseTemplatePage = await page.waitForSelector('h1');
+    const headerText = await chooseTemplatePage.textContent();
+    expect(headerText).toContain('Choose a template type to create');
+  });
+
+  test('Name link navigation - navigates to preview page', async ({
+    page,
+    baseURL,
+  }) => {
+    const manageTemplatesPage = new ManageTemplatesPage(page);
+    await manageTemplatesPage.loadPage();
+
+    expect(page.url()).toContain(`${baseURL}/templates/manage-templates`);
+
+    const templatePreviewLink = page.getByText(
+      'email-not-yet-submitted_manage-templates-page'
+    );
+
+    // This will break and need updating during CCM-7649
+    expect(templatePreviewLink).toHaveAttribute('href', '#');
+    await templatePreviewLink.click();
+    await expect(page).toHaveURL(new RegExp('/templates/manage-templates'));
+  });
+
+  test('Copy link navigation - navigates user to duplicate template type page', async ({
+    page,
+    baseURL,
+  }) => {
+    const manageTemplatesPage = new ManageTemplatesPage(page);
+    await manageTemplatesPage.loadPage();
+
+    expect(page.url()).toContain(`${baseURL}/templates/manage-templates`);
+
+    const templateRow = page.locator(
+      'tr:has-text("email-submitted_manage-templates-page")'
+    );
+    const templateCopyLink = templateRow.getByText('Copy', { exact: true });
+
+    // This will break and need updating during CCM-5539
+    expect(templateCopyLink).toHaveAttribute('href', '#');
+    await templateCopyLink.click();
+    await expect(page).toHaveURL(new RegExp('/templates/manage-templates'));
+  });
+
+  test('Delete link navigation - navigates user to delete template page', async ({
+    page,
+    baseURL,
+  }) => {
+    const manageTemplatesPage = new ManageTemplatesPage(page);
+    await manageTemplatesPage.loadPage();
+
+    expect(page.url()).toContain(`${baseURL}/templates/manage-templates`);
+
+    const templateRow = page.locator(
+      'tr:has-text("email-not-yet-submitted_manage-templates-page")'
+    );
+    const templateDeleteLink = templateRow.getByText('Delete', { exact: true });
+
+    // This will break and need updating during CCM-7572
+    expect(templateDeleteLink).toHaveAttribute('href', '#');
+    await templateDeleteLink.click();
+    await expect(page).toHaveURL(new RegExp('/templates/manage-templates'));
+  });
+
+  test('Delete link not present for submitted templates', async ({
+    page,
+    baseURL,
+  }) => {
+    const manageTemplatesPage = new ManageTemplatesPage(page);
+    await manageTemplatesPage.loadPage();
+
+    expect(page.url()).toContain(`${baseURL}/templates/manage-templates`);
+
+    const templateRow = page.locator(
+      'tr:has-text("email-submitted_manage-templates-page")'
+    );
+    const templateDeleteLink = templateRow.getByText('Delete', { exact: true });
+    await expect(templateDeleteLink).toBeHidden();
```

Comment:
Does this mean the HTML element has a CSS of `display:hidden`?

I'd expect the element to be rendered rather than adding styling to it.

Reply:
No, it's not rendered at all.
`tobeHidden()` checks if an element is hidden via styling OR not present at all.
I could change it to `await expect(templateDeleteLink).not.toBeInTheDocument();` to specifically check that it's not in the DOM?

Reply:
I'll be honest, it did the same to me. I think Jas made the change originally and I had to check it's implementation.